src/Magpie/Magpie/Services/UpdateDecider.cs

```diff
@@ -6,7 +6,7 @@
 
 namespace Magpie.Services
 {
-    internal class UpdateDecider
+    public class UpdateDecider
```

Comment:
Although this works, I think you can also get away by just extending this class and overriding the virtual methods for testing (without having to NSubstitute it)